src/next-transpile-modules.js

```diff
@@ -187,6 +188,30 @@ const withTmInitializer = (modules = [], options = {}) => {
           `**node_modules/{${modules.map((mod) => `!(${mod})`).join(',')}}/**/*`,
         ];
 
+        if (isWebpack5) {
+          const checkForTranspiledModules = (currentPath) =>
+            modules.find((mod) => {
+              return currentPath.includes(path.dirname(mod)) || currentPath.includes(mod);
+            });
+
+          const snapshot = Object.assign({}, config.snapshot);
+          const mainPkg = require(pkgUp.sync());
```

Comment:
for better caching, it might be worth traversing over the transpiled modules package.json files (the symlinked ones at least)